include/nbl/builtin/hlsl/scan/indirect.hlsl

```diff
@@ -13,36 +8,33 @@ namespace hlsl
 {
 namespace scan
 {
-#ifndef _NBL_HLSL_SCAN_GET_PARAMETERS_DEFINED_
+    
+uint32_t getIndirectElementCount();
+
 Parameters_t scanParams;
+    
 Parameters_t getParameters()
 {
-	return scanParams;
+    return scanParams;
 }
-#define _NBL_HLSL_SCAN_GET_PARAMETERS_DEFINED_
-#endif
-
-uint getIndirectElementCount();
 
-#ifndef _NBL_HLSL_SCAN_GET_SCHEDULER_PARAMETERS_DEFINED_
 DefaultSchedulerParameters_t schedulerParams;
+
 DefaultSchedulerParameters_t getSchedulerParameters()
 {
-	scheduler::computeParameters(getIndirectElementCount(),scanParams,schedulerParams);
+    scheduler::computeParameters(getIndirectElementCount(),scanParams,schedulerParams);
 	return schedulerParams;
 }
-#define _NBL_HLSL_SCAN_GET_SCHEDULER_PARAMETERS_DEFINED_
-#endif
+
 }
 }
 }
 
-#ifndef _NBL_HLSL_MAIN_DEFINED_
-[numthreads(_NBL_HLSL_WORKGROUP_SIZE_, 1, 1)]
-void CSMain()
+[numthreads(WORKGROUP_SIZE,1,1)]
+void main()
 {
-	if (bool(nbl::hlsl::scan::getIndirectElementCount()))
-		nbl::hlsl::scan::main();
+    if(bool(nbl::hlsl::scan::getIndirectElementCount())) {
+        // TODO call main from virtual_workgroup.hlsl
+    }
```

Comment:
all this should be userspace, also there will be very little difference between direct and indirect